### PR TITLE
Distribute non-blockdev functions in blockdev.rs

### PR DESCRIPTION
**Pull request by mulkieran**
These functions all primarily interact with devices not as Stratis
blockdevs, but just as devices.

Signed-off-by: mulhern <amulhern@redhat.com>

### CODE REVIEW

File: src/engine/strat_engine/device.rs [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        let bda = try!(BDA::initialize(&mut f,
+                                       pool_uuid,
+                                       &Uuid::new_v4(),
+                                       mda_size,
+                                       dev_size.sectors()));
+        let allocator = RangeAllocator::new_with_used(bda.dev_size(), &[(Sectors(0), bda.size())]);
+
+        bds.push(BlockDev::new(dev, devnode, bda, allocator));
+    }
+    Ok(bds)
```
Comment (agrover) [inline src/engine/strat_engine/device.rs]: Last 12 lines in `initialize` here are extremely BlockDev-specific. Can this stuff stay in blockdev.rs, or something like that? Given the delineation we are now drawing between device.rs and blockdev.rs, having device.rs not using `BlockDev` at all seems attractive.

Comment (mulkieran) [inline src/engine/strat_engine/device.rs]: We'ld have to change the name, too, as those are the lines that do the initializing.

**Comment by mulkieran** [comment]
New and different. @agrover, @trgill review?

**Comment by mulkieran** [comment]
I should move ```resolve_devices``` to blockdevmgr as well...

**Comment by mulkieran** [comment]
@agrover @trgill I believe it is ready now.

**Comment by mulkieran** [comment]
One last fix...

**Comment by mulkieran** [comment]
@agrover @trgill review, please?

**Comment by mulkieran** [comment]
I will start to block on this pretty soon...

**Comment by agrover** [comment]
I like it. 👍

**Pull request merged**